task/conditional_step.go

```diff
@@ -0,0 +1,58 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (C) 2016 VMware, Inc. All rights reserved.
+// -- VMware Confidential
+///////////////////////////////////////////////////////////////////////
+
+package task
+
+import (
+	"context"
+
+	"github.com/casualjim/rabbit"
+	"github.com/casualjim/rabbit/eventbus"
+)
+
+// Conditional step
+// step that executes the condFn, that returns
+// a negative integer, zero, or a positive integer
+// if the result is negative, the lStep Step is run
+// if the result is positive, the rStep Step is run
+// else no step is run
+
+type ConditionalStep struct {
+	condFn func(ctx context.Context) (context.Context, int, error)
+	lStep  Step
+	rStep  Step
+	log    rabbit.Logger
+}
+
+func NewConditionalStep(stepInfo StepInfo,
+	log rabbit.Logger,
+	lStep, rStep Step,
+	condFn func(context.Context) (context.Context, int, error)) *ConditionalStep {
```

Comment:
> why would the context change? 

the task might need to set a value for either step to use 
e.g the function checks if scale up or scale down is required and also passes the current cluster configuration along for the next steps to pickup 

> why a number 
to choose a step L or R